opensrp-reveal/src/main/java/org/smartregister/reveal/sync/LocationTaskIntentService.java

```diff
@@ -42,17 +42,22 @@ private void doSync() {
         LocationServiceHelper locationServiceHelper = LocationServiceHelper.getInstance();
         TaskServiceHelper taskServiceHelper = TaskServiceHelper.getInstance();
 
+
         List<Location> syncedStructures = locationServiceHelper.fetchLocationsStructures();
         List<Task> synchedTasks = taskServiceHelper.syncTasks();
 
-
         if (hasChangesInCurrentOperationalArea(syncedStructures, synchedTasks)) {
             Intent intent = new Intent(STRUCTURE_TASK_SYNCHED);
             LocalBroadcastManager.getInstance(getApplicationContext()).sendBroadcast(intent);
         }
 
         clientProcessEvents(extractStructureIds(syncedStructures, synchedTasks));
 
+        if (!org.smartregister.util.Utils.isEmptyCollection(syncedStructures)
+                || !org.smartregister.util.Utils.isEmptyCollection(syncedStructures)) {
```

Comment:
The conditional checks the same thing twice.

Reply:
Fixed the logical bug, was to check `synchedTasks`  and `syncedStructures`